src/compiler/binder.ts

```diff
@@ -537,6 +537,10 @@ namespace ts {
                 symbol.parent = parent;
             }
 
+            if (node.flags & NodeFlags.Deprecated) {
+                symbol.isDeprecated = true;

```

Comment:
This would be better as a symbol flag too, if there's still space (which I think there is)